File: app/src/main/java/com/mikhailovskii/trakttv/db/room/MovieIdConverter.java
package com.mikhailovskii.trakttv.db.room;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.mikhailovskii.trakttv.data.entity.MovieId;

public class MovieIdConverter {

    @TypeConverter
    public String fromId(MovieId movieId){
        return new Gson().toJson(movieId);
    }

    @TypeConverter
    public MovieId toId(String data){
        return new Gson().fromJson(data, MovieId.class);
    }

}

File: app/src/main/java/com/mikhailovskii/trakttv/db/room/MovieIdConverter.kt
package com.mikhailovskii.trakttv.db.room

import androidx.room.TypeConverter

import com.google.gson.Gson
import com.mikhailovskii.trakttv.data.entity.MovieId

class MovieIdConverter {

    @TypeConverter
    fun fromId(movieId: MovieId): String {
        return Gson().toJson(movieId)
    }

    @TypeConverter
    fun toId(data: String): MovieId {
        return Gson().fromJson(data, MovieId::class.java!!)
    }

}
